src/backend/gp_libpq_fe/fe-exec.c

```diff
@@ -2968,7 +2968,7 @@ PQoidStatus(const PGresult *res)
 
 	size_t		len;
 
-	if (!res || !res->cmdStatus || strncmp(res->cmdStatus, "INSERT ", 7) != 0)
+	if (!res || strncmp(res->cmdStatus, "INSERT ", 7) != 0)
```

Comment:
`res->cmdStatus` can not be NULL if `res` is not NULL. Please check out the definition of `PGresult`, it is a stack memory bundled with `res` variable.